lib/vtls/openssl.c

```diff
@@ -2116,7 +2124,7 @@ static CURLcode ossl_connect_step1(struct connectdata *conn, int sockindex)
   ciphers = SSL_CONN_CONFIG(cipher_list);
   if(!ciphers)
     ciphers = (char *)DEFAULT_CIPHER_SELECTION;
-  if(!SSL_CTX_set_cipher_list(BACKEND->ctx, ciphers)) {
+  if(ciphers && !SSL_CTX_set_cipher_list(BACKEND->ctx, ciphers)) {
     failf(data, "failed setting cipher list: %s", ciphers);
     return CURLE_SSL_CIPHER;
   }
```

Comment:
The line below needs to be corrected infof(data, "Cipher selection: %s\n", ciphers); but now ciphers can be NULL . Github won't let me select the line because it's too far out of the common area. There may  be a n easy way to get it in OpenSSL but I think it would be ok to just do `ciphers ? ciphers : "<default>"` or something like that

Reply:
The `if(ciphers ...` condition surely  should make sure that ciphers can't be NULL there... ?

Reply:
yeah it's confusing, github won't let me select the line, it's the line 2131

Reply:
@jay Thanks for review!  The ```infof()``` should be fixed now, will ask OpenSSL experts about a suitable version of OpenSSL for the condition.

Btw. you can refer to any location in any revision by the Github URL: https://github.com/kdudka/curl/blob/0b613865/lib/vtls/openssl.c#L2131